### PR TITLE
Python2to3: cmp to key conversion in list sort substitute

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:
Fixes python3 implementation of list sorting using a comparison function
as described here: 
https://stackoverflow.com/questions/2531952/how-to-use-a-custom-comparison-function-in-python-3
#### PR validation:
Works both with py2 and py3

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

**Comment by Dr15Jones** [comment]
How was this tested ?

**Comment by mrodozov** [comment]
This test 
TestFWCoreConcurrencyDriver 
was failing in 
https://github.com/cms-sw/cmsdist/pull/5241 
and using the PY3 IB I tried this suggestion, tried also with py2 IB and also worked.

File: FWCore/Concurrency/scripts/edmStreamStallGrapher.py [hunk 1]
```diff
@@ -424,7 +424,7 @@ def printStalledModulesInOrder(stalledModules):
 
     def sumSort(i,j):
         return cmp(i[1],j[1])
-    priorities.sort(cmp=sumSort, reverse=True)
+    priorities.sort(key=cmp_to_key(sumSort), reverse=True)
```
Comment (smuzaffar) [inline FWCore/Concurrency/scripts/edmStreamStallGrapher.py]: following should work for both py2 and 3
```
priorities = sorted(priorities , key=lambda a: a[1], reverse=True)
```

Comment (Dr15Jones) [inline FWCore/Concurrency/scripts/edmStreamStallGrapher.py]:
```suggestion
    priorities.sort(key=lambda a: a[1], reverse=True)
```
And then delete the function `sumSort`. This allows changing the (possibly large) list inplace which is probably more efficient.

**Comment by mrodozov** [comment]
please test

File: FWCore/Concurrency/scripts/edmStreamStallGrapher.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import sys
 from collections import defaultdict
 import six
-
+from functools import cmp_to_key
```
Comment (Dr15Jones) [inline FWCore/Concurrency/scripts/edmStreamStallGrapher.py]: The import does not appear to be needed

**Comment by mrodozov** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**